src/k8s/pkg/k8sd/api/capi_access_handler.go

```diff
@@ -11,28 +11,28 @@ import (
 	"github.com/canonical/microcluster/state"
 )
 
-func ValidateCAPIAuthTokenAccessHandler(tokenHeaderName string) func(s *state.State, r *http.Request) response.Response {
-	return func(s *state.State, r *http.Request) response.Response {
+func ValidateCAPIAuthTokenAccessHandler(tokenHeaderName string) func(s state.State, r *http.Request) (bool, response.Response) {
+	return func(s state.State, r *http.Request) (bool, response.Response) {
 		token := r.Header.Get(tokenHeaderName)
 		if token == "" {
-			return response.Unauthorized(fmt.Errorf("missing header %q", tokenHeaderName))
+			return false, response.Unauthorized(fmt.Errorf("missing header %q", tokenHeaderName))
 		}
 
 		var tokenIsValid bool
-		if err := s.Database.Transaction(s.Context, func(ctx context.Context, tx *sql.Tx) error {
+		if err := s.Database().Transaction(r.Context(), func(ctx context.Context, tx *sql.Tx) error {
```

Comment:
I noticed we are changing many instance `s/s.Context/r.Context()`. What's the reason for this change?

Reply:
Please see the MM link above for the full story.

Tl;dr: Microcluster does not expose `s.Context` any more and for *most* of the usages that we had, the request context is perfectly fine.